examples/experimental/dagster-airlift/dagster_airlift/dbt/multi_asset.py

```diff
@@ -1,37 +1,91 @@
-import json
 from dataclasses import dataclass
-from pathlib import Path
-from typing import Optional
+from typing import Any, Mapping, Optional
 
-from dagster import AssetExecutionContext, Definitions
-from dagster_dbt import DbtCliResource, dbt_assets
+from dagster import AssetExecutionContext, Definitions, multi_asset
+from dagster_dbt import (
+    DagsterDbtTranslator,
+    DbtCliResource,
+    DbtProject,
+    build_dbt_asset_specs,
+    dbt_assets,
+)
+from dagster_dbt.dbt_manifest import DbtManifestParam, validate_manifest
 
 from dagster_airlift.core import DefsFactory
 
 
 @dataclass
 class DbtProjectDefs(DefsFactory):
-    dbt_project_path: Path
+    """A factory that builds a :py:class:`dagster.Definitions` object from a dbt project.
+    If the dbt project is not living within the dagster codebase and/or the dbt project is not
+    being orchestrated by dagster, just provide a reference to the manifest and external assets
+    will be constructed.
+
+    Args:
+        dbt_manifest (DbtManifestParam): The dbt manifest. This can be a path to a manifest file,
+            a string of the manifest JSON, or the parsed manifest JSON.
+        dbt_project_path (Path): The path to the dbt project.
+        name (str): The name to give the dbt project. In the case of airflow-orchestrated DBT assets,
+            this should be <dag_id>__<task_id>.
+        group (Optional[str], optional): The asset group name for the dbt assets. Default is "default".
+    """
+
+    dbt_manifest: Mapping[str, Any]
     name: str
-    group: Optional[str] = None
+    translator: Optional[DagsterDbtTranslator]
```

Comment:
> Allows for more flexible DbtProjectDefs

Is there a particular motivating example for this? I'm trying to discern if the complexity is necessary for migration or if the main goal of this PR is to support further customization post-migration (e.g. once I'm done migrating, I can customize my specs through the Translator).

If it's the latter, I wonder if we should prompt users to move off the `DefsFactory`. Seems like there is some risk of being forced to lift complexity of lower-level APIs (like `@dbt_assets`) into higher level APIs (`DefsFactory` layer). Now we need to document all of the arguments (translators, `DbtProject` input etc) in two places. We already hit this with e.g. `@dbt_assets` replicating complexity of `@multi_asset`.

If the answer is "some level of customization is necessary even for basic migration" (which is probably the case), and we do need to replicate some of the underlying dbt complexity into these factories, I think we should be judicious & keep it as minimal as possible.

Reply:
I see your point, I don't mind getting rid of translator arg here. I think the main thing to support is just make sure selection/exclude are supported out of the box. I think the goal should be anything you can easily accomplish easily with the bash operator/dbt CLI should be covered by an API like this.

I think it's worth revisiting the top-level API for any marketing event/real release. But I think the focus rn is just making sure to unblock patterns, and figure out the nice API later once we understand the full scope of patterns people want.

Reply:
> I think the goal should be anything you can easily accomplish easily with the bash operator/dbt CLI should be covered by an API like this.

cool, this seems like a good dividing line to me

> But I think the focus rn is just making sure to unblock patterns

For sure, don't see an issue in merging as-is